test/view/shared.cpp

```diff
@@ -0,0 +1,294 @@
+        CHECK(vec_view.size() == 5u);
+        CHECK(vec_view.get_shared().use_count() == 2);
+        ::check_equal(vec_view, {1, 2, 8, 8, 8});
+    }
+
+    {
+        // check the piped construction from an rvalue of a shared_ptr
+        auto base_vec_ptr = std::make_shared<std::vector<int>>();
+        *base_vec_ptr = {1, 2, 8, 8, 8};
+        auto vec_view = std::move(base_vec_ptr) | view::shared;
+        CHECK(vec_view.size() == 5u);
+        CHECK(vec_view.get_shared().use_count() == 1);
+        ::check_equal(vec_view, {1, 2, 8, 8, 8});
+    }
+
+    {
+        // test bidirectional range
+        auto list_view = std::list<int>{1, 2, 3} | view::shared;
+
+        CHECK(list_view.size() == 3u);
+        has_type<int &>(*begin(list_view));
+        ::models<concepts::SizedRange>(list_view);
+        ::models<concepts::BidirectionalRange>(list_view);
+        ::models<concepts::BidirectionalIterator>(begin(list_view));
+        ::models_not<concepts::RandomAccessRange>(list_view);
+        ::models_not<concepts::RandomAccessIterator>(begin(list_view));
+
+        // test bidirectional range iterator
+        auto list_iter = begin(list_view);
+        CHECK(*list_iter == 1);
+        CHECK(*(++list_iter) == 2);
+        auto nxt = next(list_iter, 1);
+        auto prev = next(list_iter, -1);
+        CHECK(*nxt == 3);
+        CHECK(*prev == 1);
+        CHECK(*(--nxt) == *list_iter);
+        CHECK(*(++prev) == *list_iter);
+        CHECK(*(--list_iter) == 1);
+    }
+
+    {
+        // test bidirectional range, which does not know its size
+        auto base_list = std::list<int>{1, 2, 3};
+        auto iter_range = make_iterator_range(begin(base_list), end(base_list));
+        auto list_view = std::move(iter_range) | view::shared;
+
+        has_type<int &>(*begin(list_view));
+        ::models_not<concepts::SizedRange>(list_view);
+        ::models<concepts::BidirectionalRange>(list_view);
+        ::models<concepts::BidirectionalIterator>(begin(list_view));
+
+        // test its iterator
+        auto list_iter = begin(list_view);
+        CHECK(*list_iter == 1);
+        CHECK(*(++list_iter) == 2);
+        CHECK(*(--list_iter) == 1);
+    }
+
+    {
+        // test random access range
+        auto vec_view = std::vector<int>{1, 2, 3} | view::shared;
+
+        CHECK(vec_view.size() == 3u);
+        has_type<int &>(*begin(vec_view));
+        ::models<concepts::SizedRange>(vec_view);
+        ::models<concepts::RandomAccessRange>(vec_view);
+        ::models<concepts::RandomAccessIterator>(begin(vec_view));
+        CHECK(vec_view[0] == 1);
+        CHECK(vec_view[1] == 2);
+        CHECK(vec_view[2] == 3);
+
+        // test random access range iterator
+        auto vec_iter = begin(vec_view);
+        CHECK(*vec_iter == 1);
+        CHECK(vec_iter[0] == 1);
+        CHECK(vec_iter[1] == 2);
+        CHECK(vec_iter[2] == 3);
+    }
+
+    {
+        // test const random access range
```

Comment:
What coverage does this provide over the previous test? Nothing in the view definition is sensitive to the constness of the range type or its elements.